edgelet/edgelet-http/src/util/hyperwrap.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use error::Error;
+use futures::future;
+use hyper::client::{HttpConnector, Service};
+use hyper::{Client as HyperClient, Error as HyperError, Request, Response, StatusCode, Uri};
+use hyper_proxy::{Intercept, Proxy, ProxyConnector};
+use hyper_tls::HttpsConnector;
+use tokio_core::reactor::Handle;
+
+const DNS_WORKER_THREADS: usize = 4;
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    handle: Option<Handle>,
+    proxy_uri: Option<Uri>,
+    null: bool,
+}
+
+impl Config {
+    pub fn handle<'a>(&'a mut self, handle: &Handle) -> &'a mut Config {
+        self.handle = Some(handle.clone());
+        self
+    }
+
+    pub fn proxy<'a>(&'a mut self, uri: Uri) -> &'a mut Config {
+        self.proxy_uri = Some(uri);
+        self
+    }
+
+    pub fn null<'a>(&'a mut self) -> &'a mut Config {
+        self.null = true;
+        self
+    }
+
+    pub fn build(&self) -> Result<Client, Error> {
+        match self.null {
+            true => Ok(Client::Null),
+            false => {
+                let config = self.clone();
+                let h = &config
+                    .handle
+                    .expect("tokio_core::reactor::Handle expected!");
```

Comment:
Is expect recommended? Or is it okay here?

Reply:
It's what we want in this case.

In a production environment there must be a handle at this point or else we can go no further. The question, then, would be whether to panic or return an error. Because this is an internal component, and because its consumer (`MaybeProxyClient`) always passes in the handle (the handle is a required argument to `MaybeProxyClient::new`), we don't want to mess with new error codes and extra handling; we just want to fail fast.

If some developer introduced a bug in the future, e.g., removed the code in `MaybeProxyClient` that passes the handle, or created a 2nd consumer that didn't pass the handle, we'd know right away, long before any release.